src/jdk.compiler/share/classes/com/sun/tools/javac/jvm/Gen.java

```diff
@@ -1505,15 +1505,16 @@ void afterBody() {
 
         };
         tryEnv.info.gaps = new ListBuffer<>();
-        genTry(tree.body, tree.catchers, tryEnv);
+        genTry(tree.body, tree.catchers, tryEnv, true);
     }
     //where
         /** Generate code for a try or synchronized statement
          *  @param body      The body of the try or synchronized statement.
-         *  @param catchers  The lis of catch clauses.
-         *  @param env       the environment current for the body.
+         *  @param catchers  The list of catch clauses.
+         *  @param env       The current environment of the body.
+         *  @param actualTry Identify try or synchronized statement, true for try and false for synchronized.
          */
-        void genTry(JCTree body, List<JCCatch> catchers, Env<GenContext> env) {
+        void genTry(JCTree body, List<JCCatch> catchers, Env<GenContext> env, boolean actualTry) {
```

Comment:
I don't think you really need an additional argument, I think that you can check for the tree at the `env` argument and you should be able to find out if this method is being invoked from a `try` or not. Kind of:
```
boolean actualTry = env.tree.hasTag(TRY);
```

Reply:
@vicente-romero-oracle you are right. I updated the code just now by using the `env` instead of the new parameter.